File: decompiled/sources/com/hjq/widget/view/ClearEditText.java
package com.hjq.widget.view;

import android.content.Context;
import android.graphics.drawable.Drawable;
import android.text.Editable;
import android.text.TextWatcher;
import android.util.AttributeSet;
import android.view.MotionEvent;
import android.view.View;
import androidx.core.content.ContextCompat;
import androidx.core.graphics.drawable.DrawableCompat;
import com.hjq.widget.R;

public final class ClearEditText extends RegexEditText implements View.OnTouchListener, View.OnFocusChangeListener, TextWatcher {
    private final Drawable mClearDrawable;
    private View.OnFocusChangeListener mFocusChangeListener;
    private View.OnTouchListener mTouchListener;

    public void afterTextChanged(Editable editable) {
    }

    public void beforeTextChanged(CharSequence charSequence, int i, int i2, int i3) {
    }

    public ClearEditText(Context context) {
        this(context, (AttributeSet) null);
    }

    public ClearEditText(Context context, AttributeSet attributeSet) {
        this(context, attributeSet, 16842862);
    }

    public ClearEditText(Context context, AttributeSet attributeSet, int i) {
        super(context, attributeSet, i);
        Drawable wrap = DrawableCompat.wrap(ContextCompat.getDrawable(context, R.drawable.input_delete_ic));
        this.mClearDrawable = wrap;
        wrap.setBounds(0, 0, wrap.getIntrinsicWidth(), wrap.getIntrinsicHeight());
        setDrawableVisible(false);
        super.setOnTouchListener(this);
        super.setOnFocusChangeListener(this);
        super.addTextChangedListener(this);
    }

    private void setDrawableVisible(boolean z) {
        if (this.mClearDrawable.isVisible() != z) {
            this.mClearDrawable.setVisible(z, false);
            Drawable[] compoundDrawablesRelative = getCompoundDrawablesRelative();
            setCompoundDrawablesRelative(compoundDrawablesRelative[0], compoundDrawablesRelative[1], z ? this.mClearDrawable : null, compoundDrawablesRelative[3]);
        }
    }

    public void setOnFocusChangeListener(View.OnFocusChangeListener onFocusChangeListener) {
        this.mFocusChangeListener = onFocusChangeListener;
    }

    public void setOnTouchListener(View.OnTouchListener onTouchListener) {
        this.mTouchListener = onTouchListener;
    }

    public void onFocusChange(View view, boolean z) {
        boolean z2 = false;
        if (!z || getText() == null) {
            setDrawableVisible(false);
        } else {
            if (getText().length() > 0) {
                z2 = true;
            }
            setDrawableVisible(z2);
        }
        View.OnFocusChangeListener onFocusChangeListener = this.mFocusChangeListener;
        if (onFocusChangeListener != null) {
            onFocusChangeListener.onFocusChange(view, z);
        }
    }

    public boolean onTouch(View view, MotionEvent motionEvent) {
        boolean z;
        int x = (int) motionEvent.getX();
        int layoutDirection = getLayoutDirection();
        if (layoutDirection != 0 ? layoutDirection != 1 || x <= getPaddingStart() || x >= getPaddingStart() + this.mClearDrawable.getIntrinsicWidth() : x <= (getWidth() - this.mClearDrawable.getIntrinsicWidth()) - getPaddingEnd() || x >= getWidth() - getPaddingEnd()) {
            z = false;
        } else {
            z = true;
        }
        if (!this.mClearDrawable.isVisible() || !z) {
            View.OnTouchListener onTouchListener = this.mTouchListener;
            if (onTouchListener == null || !onTouchListener.onTouch(view, motionEvent)) {
                return false;
            }
            return true;
        }
        if (motionEvent.getAction() == 1) {
            setText("");
        }
        return true;
    }

    public void onTextChanged(CharSequence charSequence, int i, int i2, int i3) {
        if (isFocused()) {
            setDrawableVisible(charSequence.length() > 0);
        }
    }
}

File: decompiled/sources/com/hjq/widget/view/ClearEditText.kt
package com.hjq.widget.view

import android.content.Context
import android.graphics.drawable.Drawable
import android.text.Editable
import android.text.TextWatcher
import android.util.AttributeSet
import android.view.MotionEvent
import android.view.View
import androidx.core.content.ContextCompat
import androidx.core.graphics.drawable.DrawableCompat
import com.hjq.widget.R

class ClearEditText : RegexEditText, View.OnTouchListener, View.OnFocusChangeListener, TextWatcher {
    private val mClearDrawable: Drawable
    private var mFocusChangeListener: View.OnFocusChangeListener? = null
    private var mTouchListener: View.OnTouchListener? = null

    constructor(context: Context) : this(context, null)

    constructor(context: Context, attrs: AttributeSet?) : this(context, attrs, 16842862)

    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : super(
        context,
        attrs,
        defStyleAttr
    ) {
        val wrap = DrawableCompat.wrap(ContextCompat.getDrawable(context, R.drawable.input_delete_ic)!!)
        mClearDrawable = wrap
        wrap.setBounds(0, 0, wrap.intrinsicWidth, wrap.intrinsicHeight)
        setDrawableVisible(false)
        super.setOnTouchListener(this)
        super.setOnFocusChangeListener(this)
        super.addTextChangedListener(this)
    }

    private fun setDrawableVisible(visible: Boolean) {
        if (mClearDrawable.isVisible != visible) {
            mClearDrawable.setVisible(visible, false)
            val drawables = compoundDrawablesRelative
            setCompoundDrawablesRelative(
                drawables[0],
                drawables[1],
                if (visible) mClearDrawable else null,
                drawables[3]
            )
        }
    }

    override fun setOnFocusChangeListener(listener: View.OnFocusChangeListener?) {
        mFocusChangeListener = listener
    }

    override fun setOnTouchListener(listener: View.OnTouchListener?) {
        mTouchListener = listener
    }

    override fun onFocusChange(view: View, hasFocus: Boolean) {
        if (!hasFocus || text == null) {
            setDrawableVisible(false)
        } else {
            setDrawableVisible(text!!.isNotEmpty())
        }
        mFocusChangeListener?.onFocusChange(view, hasFocus)
    }

    override fun onTouch(view: View, event: MotionEvent): Boolean {
        val x = event.x.toInt()
        val layoutDirection = layoutDirection
        val touched =
            if (layoutDirection != 0) {
                layoutDirection == 1 &&
                    x > paddingStart &&
                    x < paddingStart + mClearDrawable.intrinsicWidth
            } else {
                x > (width - mClearDrawable.intrinsicWidth) - paddingEnd &&
                    x < width - paddingEnd
            }
        if (!mClearDrawable.isVisible || !touched) {
            val onTouchListener = mTouchListener
            if (onTouchListener == null || !onTouchListener.onTouch(view, event)) {
                return false
            }
            return true
        }
        if (event.action == MotionEvent.ACTION_UP) {
            setText("")
        }
        return true
    }

    override fun beforeTextChanged(s: CharSequence, start: Int, count: Int, after: Int) {
    }

    override fun onTextChanged(s: CharSequence, start: Int, before: Int, count: Int) {
        if (isFocused) {
            setDrawableVisible(s.isNotEmpty())
        }
    }

    override fun afterTextChanged(s: Editable) {
    }
}
